### DISCUSSION TITLE
Get products from filtered collections with facets is very slow ~ 1.3s

### DISCUSSION THREAD

**Discussion by bitforcesrl** [Shopify/storefront-api-feedback#167] [Help]
Hi all, we are using, insde an hydrogen project, a query to  show collections products, facets and apply filters.
The filter works well but the speed to return items it's seems very slow. ~1.3 seconds.
we are asking for just 20 products and for every produts we ask for only 1 variant.
We've notified this issue to the Hydrogen team too.
Here you can find a video of me uncommenting progressively the below query: https://recordit.co/jabIAi74Ek

The query, nothing fancy
```
fragment ProductCard on Product {
  id
  title
  publishedAt
  handle
  options {
    id
    name
    values
  }
  variants(first: 1) {
    nodes {
      id
      image {
        url
        altText
        width
        height
      }
      priceV2 {
        amount
        currencyCode
      }
      compareAtPriceV2 {
        amount
        currencyCode
      }
    }
  }
}
query CollectionDetails(
  $handle: String!
  $country: CountryCode
  $language: LanguageCode
  $pageBy: Int!
  $cursor: String
  $productFilters: [ProductFilter!] = []
  $sortKey: ProductCollectionSortKeys
  $reverse: Boolean = false
) @inContext(country: $country, language: $language) {
  collection(handle: $handle) {
    id
    title
    description
    seo {
      description
      title
    }
    image {
      id
      url
      width
      height
      altText
    }
    products(
      first: $pageBy
      after: $cursor
      filters: $productFilters
      sortKey: $sortKey
      reverse: $reverse
    ) {
      nodes {
        ...ProductCard
      }
      filters {
        id
        label
        type
        values {
          id
          label
          count
          input
        }
      }
      pageInfo {
        hasNextPage
        endCursor
      }
    }
  }
}
```

Varables
```
{"handle": "capispalla","country": "IT", "language": "IT","pageBy": 20, 
  "productFilters":[{"productVendor": "Kocca"}]}
```

**Comment by mdunbavan** [comment]
@bitforcesrl this is something we are using an external provider for called Klevu who handle filtering.
This comes with huge budget considerations though so something like this would be brilliant if it worked as fast as an external service provider such as Klevu or Algolia.

**Reply by bitforcesrl** [reply]
We are using algolia for some clients too, but it's exepensive and overkill for some easy shops.
The Shopify standard implementation of filtering should be enough for several clients and evetyally you can upgrade to Algolia or similar.
you can check our demo here: https://www.bitforce.shop/collections/capispalla

The point is that now is slow and facets are not translated

**Reply by johncraigcole** [reply]
Hey @bitforcesrl taking a look. I'll follow up when I have more to share.

**Reply by mdunbavan** [reply]
This would be amazing to get information and some timeline of investigation. A client of ours and a big one in the UK is looking at filtering with facets and speed is absolutely crucial to it. We use Klevu right now but it’s just not fit for purpose and isn’t great for getting the correct currency back in items sent back to the frontend. If Shopify could look at speed and translations with this query it would be a game changer.

**Reply by bitforcesrl** [reply]
@mdunbavan  this is somethig that they wil address i suppose. 1.3s to get a simple list of products with filters it's super long and destroys all the effort to go headless! let's hope that someone will take care of this item soon.
if you are on hydrogen too please contribute to this list of missing things needed before to go in production!

**Reply by johncraigcole** [reply]
Hey @bitforcesrl looks like you're using `unstable` on the shop you referenced. We've been doing some incredible Storefront API work in 2022 and into 2023 to get queries served at the edge. Using `unstable` negated that from happening. That being said, `2023-01` is not fully supported yet, but if you use `2022-10` for now, you should see a huge improvement.

**Reply by bitforcesrl** [reply]
yep `2022-10` runs in around 350ms / 400ms lot faster. 
So is `2022-10` running in the edge or not?
Thanks for the hint! When 2023-01 or unstable will be upated?

**Reply by johncraigcole** [reply]
In this instance of a Hydrogen storefront which I believe is deployed on a Cloudflare worker,`2022-10` would be served from the same region as the worker. A request against the `unstable` version or `2023-01` have different handling and routing at this time. The `2023-01` version should see similar support to `2022-10` shortly, but `unstable` requests I wouldn't expect the same support for soon. I'll of course follow up when I know more.

**Reply by bitforcesrl** [reply]
I was testing it from my pc and not reading those timings from the hydrogen instance, i'm not located close to the API (from Italy)
The 2022-10 version is lot faster (so those are served at the edge, close to me) but it's missing some api's that we are using in the hydrogen commerce. i'll not spend time to check what.

So to recap 2022-10 is running at the edge and starting from 2023-01 the api should be modified to be served fom the edge right?

We'll wait the "new" edge support coming for 2023-01

**Reply by johncraigcole** [reply]
Yes your understanding is correct. We took an iterative approach to this work of getting Storefront API request served at the edge, so apologize for the inconsistency experienced with `unstable` and `2023-01`. That being said, when this work is complete the issue experienced with `2023-01` shouldn't continue when a new API version is released. We definitely would not recommend running production against `unstable`. I know at a time earlier in the Hydrogen project we were using `unstable` but it's not something you'd want to do in production.

**Reply by bitforcesrl** [reply]
Thanks for all the details! You're right it's not a good idea to go in prod with unstable
We are still testing with our demo the overall solution as you could remember form our last meet call we still want to push an Hydro store in prod asap. 
This is a discussion that we are maintaining with things that we miss before to go in prod!
https://github.com/Shopify/hydrogen/discussions/2372

**Reply by johncraigcole** [reply]
`2023-01` Now available at the edge 🎉 🎊
